multi/imxrt-multi/config.h

```diff
@@ -219,37 +703,338 @@
 #endif
 #endif
 
+
 /* SPI */
 
 #ifndef SPI1
-#define SPI1 1
+#define SPI1 0
+#elif !ISBOOLEAN(SPI1)
+#error "SPI1 must have a value of 0, 1, or be undefined"
+#endif
```

Comment:
In final solution I would move  error/warning messages and checkers into .c file. In case of warning you will  have wall of warnings. 

However I know that currently it may be hard to achieve because of deprecation of old approach.